File: android/src/main/java/com/dodo/flutterbridge/call/InvokeAttachable.kt
package com.dodo.flutterbridge.call

/**
 *     author : liuduo
 *     e-mail : liuduo@gyenno.com
 *     time   : 2022/08/17
 *     desc   : 可以挂载父invoker
 *     version: 1.0
 */
interface InvokeAttachable<P> {

    /**
     * 挂载invoker
     * @param invoker Invoker<P>
     */
    fun attachInvoker(invoker: Invoker<P>)

    /**
     * 卸载invoker
     * @param invoker Invoker<P>
     */
    fun detachInvoker(invoker: Invoker<P>)
}

File: lib/src/call/invoke_attachable.dart
import 'package:flutter_bridge/src/call/invoker.dart';

///   author : liuduo
///   e-mail : liuduo@gyenno.com
///   time   : 2022/08/18
///   desc   : 可以挂载父invoker
///   version: 1.0
abstract class InvokeAttachable<P> {

  ///挂载invoker
  void attachInvoker(Invoker<P> invoker);

  ///卸载invoker
  void detachInvoker(Invoker<P> invoker);
}
